### PR TITLE
Doc/move database setup instructions in install md

### PR DESCRIPTION
**Pull request by fossterer**
Hello,

After setting up synapse server for the first time and creating 2 users, I was puzzled to find that no table `users` existed.
I didn't even realize I just installed but failed to change the the value for `db` in config as I originally skipped that section since I didn't even have a `homeserver.yaml` by that point.

To fix that, I reordered the section to appear after having that `homeserver.yaml` available.
Also, added syntax highlighting and more markdown cleanup.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Shashank Sabniveesu <shashank@linux.com>

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -392,20 +395,35 @@ and mounting it to `/var/synapse` should be taken into consideration.
...
-https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/matrix-synapse.nix
+<https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/matrix-synapse.nix>
 
-# Setting up Synapse
+## Setting up Synapse
 
 Once you have installed synapse as above, you will need to configure it.
 
-## TLS certificates
+### Switching the database engine
```
Comment (anoadragon453) [inline INSTALL.md:411]:
```suggestion
### Using PostgreSQL
```

File: INSTALL.md [hunk 2]
```diff
@@ -392,20 +395,35 @@ and mounting it to `/var/synapse` should be taken into consideration.
...
+
+By default Synapse gets setup using [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.
```
Comment (anoadragon453) [inline INSTALL.md:413]:
```suggestion
By default Synapse uses [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.
```

File: INSTALL.md [hunk 3]
```diff
@@ -392,20 +395,35 @@ and mounting it to `/var/synapse` should be taken into consideration.
...
+SQLite is only recommended in Synapse for testing purposes or for servers with
+light workloads.
+
+Almost all installations should opt to use [PostgreSQL](https://www.postgresql.org). Advantages include:
+
+- significant performance improvements due to the superior threading and
+  caching model, smarter query optimiser
+- allowing the DB to be run on separate hardware
+
+For information on how to install and use PostgreSQL, please see
```
Comment (anoadragon453) [inline INSTALL.md:423]:
```suggestion
For information on how to install and use PostgreSQL in Synapse, please see
```

File: INSTALL.md [hunk 4]
```diff
@@ -497,11 +516,11 @@ correctly. `public_baseurl` should be set to the URL that clients will use to
 connect to your server. This is the same URL you put for the `m.homeserver`
 `base_url` above.
 
-```
+```nginx
```
Comment (anoadragon453) [inline INSTALL.md:519]:
```suggestion
```yaml
```

This is supposed to happen in `homeserver.yaml`, per the paragraph above.

File: changelog.d/8987.doc [hunk 5]
```diff
@@ -0,0 +1 @@
+Moved instructions for database setup and adjusted heading levels and syntax highlighting in [INSTALL.md](../INSTALL.md). Contributed by fossterer.
```
Comment (anoadragon453) [inline changelog.d/8987.doc:1]:
```suggestion
Moved instructions for database setup, adjusted heading levels and improved syntax highlighting in [INSTALL.md](../INSTALL.md). Contributed by fossterer.
```

File: INSTALL.md [hunk 6]
```diff
@@ -392,20 +395,35 @@ and mounting it to `/var/synapse` should be taken into consideration.
...
+## Setting up Synapse
 
 Once you have installed synapse as above, you will need to configure it.
 
-## TLS certificates
+### Switching the database engine
+
+By default Synapse gets setup using [SQLite](https://sqlite.org/) and in doing so trades performance for convenience.
+SQLite is only recommended in Synapse for testing purposes or for servers with
+light workloads.
```
Comment (anoadragon453) [inline INSTALL.md:415]:
```suggestion
very light workloads.
```

Just for that extra bit of convincing.

**Review by anoadragon453 — APPROVED** [review]
Thank you! Lots of great improvements in here. I walked through the guide as you directed and indeed noticed that changes to `homeserver.yaml` were asked of in the postgres docs before `homeserver.yaml` exists.

I tweaked the wording a little bit, but on the whole it looks great.

File: INSTALL.md [hunk 7]
```diff
@@ -1,19 +1,44 @@
...
+      - [Fedora](#fedora)
+      - [OpenSUSE](#opensuse-1)
+      - [SUSE Linux Enterprise Server](#suse-linux-enterprise-server)
+      - [ArchLinux](#archlinux-1)
+      - [Void Linux](#void-linux)
+      - [FreeBSD](#freebsd)
+      - [OpenBSD](#openbsd-1)
+      - [NixOS](#nixos)
+  - [Setting up Synapse](#setting-up-synapse)
+    - [Switching the database engine](#switching-the-database-engine)
```
Comment (anoadragon453) [inline INSTALL.md:32]:
```suggestion
    - [Using PostgreSQL](#using-postgresql)
```

Oops, missed a change here.

**Review by anoadragon453 — COMMENTED** [review]
Thank you for your contribution!

File: INSTALL.md [hunk 8]
```diff
@@ -408,19 +408,19 @@ Robin Lambertz has packaged Synapse for NixOS at:
...
+very light workloads.
 
 Almost all installations should opt to use [PostgreSQL](https://www.postgresql.org). Advantages include:
 
 - significant performance improvements due to the superior threading and
   caching model, smarter query optimiser
 - allowing the DB to be run on separate hardware
 
-For information on how to install and use PostgreSQL, please see
+For information on how to install and use PostgreSQL in Synapse, please see
```
Comment (fossterer) [inline INSTALL.md:423]: Aah! Even better

**Pull request merged**